src/ServiceControl.Transports.PostgreSql/PostgreSqlTransportCustomization.cs

```diff
@@ -0,0 +1,86 @@
+
+    protected override void CustomizeTransportForAuditEndpoint(EndpointConfiguration endpointConfiguration, PostgreSqlTransport transportDefinition, TransportSettings transportSettings) =>
+        transportDefinition.TransportTransactionMode = TransportTransactionMode.ReceiveOnly;
+
+    protected override void CustomizeTransportForMonitoringEndpoint(EndpointConfiguration endpointConfiguration, PostgreSqlTransport transportDefinition, TransportSettings transportSettings) =>
+        transportDefinition.TransportTransactionMode = TransportTransactionMode.ReceiveOnly;
+
+    protected override void AddTransportForPrimaryCore(IServiceCollection services,
+        TransportSettings transportSettings) =>
+        services.AddSingleton<IBrokerThroughputQuery, PostgreSqlQuery>();
+
+    protected override void AddTransportForMonitoringCore(IServiceCollection services, TransportSettings transportSettings)
+    {
+        services.AddSingleton<IProvideQueueLength, QueueLengthProvider>();
+        services.AddHostedService(provider => provider.GetRequiredService<IProvideQueueLength>());
+    }
+
+    protected override PostgreSqlTransport CreateTransport(TransportSettings transportSettings, TransportTransactionMode preferredTransactionMode = TransportTransactionMode.ReceiveOnly)
+    {
+        var connectionString = transportSettings.ConnectionString.RemoveCustomConnectionStringParts(out var customSchema, out var subscriptionsTableSetting);
+
+        var transport = new PostgreSqlTransport(connectionString);
+
+        var subscriptions = transport.Subscriptions;
+
+        if (customSchema != null)
+        {
+            transport.DefaultSchema = customSchema;
+            subscriptions.SubscriptionTableName = new SubscriptionTableName(DefaultSubscriptionTableName, customSchema);
+        }
+
+        if (subscriptionsTableSetting != null)
+        {
+            var subscriptionsAddress = QueueAddress.Parse(subscriptionsTableSetting);
+
+            subscriptions.SubscriptionTableName =
+                new SubscriptionTableName(subscriptionsAddress.Table,
+                    subscriptionsAddress.Schema ?? customSchema);
+        }
+
+        if (transportSettings.GetOrDefault<bool>("TransportSettings.EnableDtc"))
+        {
+            Logger.Error("The EnableDtc setting is no longer supported natively within ServiceControl. If you require distributed transactions, you will have to use a Transport Adapter (https://docs.particular.net/servicecontrol/transport-adapter/)");
+        }
+
+        DisableDelayedDelivery(transport) = true;
+
+        transport.TransportTransactionMode = transport.GetSupportedTransactionModes().Contains(preferredTransactionMode) ? preferredTransactionMode : TransportTransactionMode.ReceiveOnly;
+
+        return transport;
+    }
+
+    protected override string ToTransportQualifiedQueueNameCore(string queueName)
+    {
+        const string delimiter = "\"";
```

Comment:
is this the same qualification scheme used in `PostgreSqlNameHelper`?

Reply:
yes - are you thinking of making those public and using them here?

Reply:
makes more sense than duplicating code